rate.go

```diff
@@ -40,3 +45,22 @@ func (c *Client) GetRateWithContext(ctx context.Context, rateID string) (out *Ra
 	err = c.get(ctx, "rates/"+rateID, &out)
 	return
 }
+
+type getShipmentSmartratesResponse struct {
+	Smartrate *[]*Smartrate `json:"result,omitempty"`
+}
+
+// GetShipmentSmartrates fetches the available smartrates for a shipment.
+func (c *Client) GetShipmentSmartrates(shipmentID string) (out []*Smartrate, err error) {
```

Comment:
Since we are defining these on a shipment, they should be defined in `shipment.go` like the other shipment dependent funcs

Reply:
Makes sense